### PR TITLE
Remove AdapterWorkAction used by WorkerExecutor

### PR DESCRIPTION
**Pull request by tresat**
Followup to #21736 

- Remove no longer necessary worker action adaptor classes
- Simplify test

### CODE REVIEW

**Comment by tresat** [comment]
@bot-gradle test this

File: subprojects/workers/src/main/java/org/gradle/workers/internal/DefaultWorkerExecutor.java [hunk 1]
```diff
@@ -183,16 +182,7 @@ private AsyncWorkCompletion submitWork(IsolatedParametersActionExecutionSpec<?>
     }
 
     private static String getWorkerDisplayName(Class<?> workActionClass, WorkParameters parameters) {
```
Comment (jvandort) [inline subprojects/workers/src/main/java/org/gradle/workers/internal/DefaultWorkerExecutor.java:184]: Can we just remove this method entirely and inline it now?

**Review by jvandort — CHANGES_REQUESTED** [review]
Seems like we can also get rid of `org.gradle.workers.WorkerConfiguration` and `org.gradle.workers.internal.DefaultWorkerConfiguration`

File: subprojects/workers/src/main/java/org/gradle/workers/internal/DefaultWorkerExecutor.java (hunk 1 above)
Comment (tresat) [reply subprojects/workers/src/main/java/org/gradle/workers/internal/DefaultWorkerExecutor.java:184]: Yes, good catch.

**Comment by tresat** [comment]
> Seems like we can also get rid of `org.gradle.workers.WorkerConfiguration` and `org.gradle.workers.internal.DefaultWorkerConfiguration`

We can, I have a separate item for those.  Trying to keep each of these removals relatively small and self-contained.

**Comment by tresat** [comment]
@bot-gradle test this

**Review by jvandort — APPROVED** [review]

**Comment by tresat** [comment]
@bot-gradle test and merrge

**Comment by tresat** [comment]
@bot-gradle test and merge

**Pull request merged**